File: apps/agni-server/src/core/interactions/notification/deleteNotification.ts
import Repository from "@core/adapters/repository";
import { IUsecase } from "../interfaces";
import Notification from "@core/domains/entities/notification";

export class DeleteNotifcationUseCase implements IUsecase<string, void> {
    private notificationRepo: Repository<Notification>

    constructor(notificationRepo: Repository<Notification>) {
        this.notificationRepo = notificationRepo
    }

    async execute(id: string): Promise<void> {
        if (await this.notificationRepo.get(id))
            await this.notificationRepo.delete(id)
    }
}

File: apps/agni_api/src/main/kotlin/dev/auguste/agni_api/core/usecases/notifications/DeleteNotification.kt
package dev.auguste.agni_api.core.usecases.notifications

import dev.auguste.agni_api.core.adapters.repositories.IRepository
import dev.auguste.agni_api.core.entities.DomainException
import dev.auguste.agni_api.core.entities.Notification
import dev.auguste.agni_api.core.usecases.interfaces.IUseCase
import dev.auguste.agni_api.core.usecases.notifications.dto.DeleteNotificationInput
import java.util.UUID

class DeleteNotification(private val notificationRepo: IRepository<Notification>): IUseCase<DeleteNotificationInput, Unit> {

    override fun execAsync(input: DeleteNotificationInput) {
        notificationRepo.get(input.notificationId) ?: throw DomainException.NotFound.Notification(input.notificationId)

        notificationRepo.delete(input.notificationId)
    }
}